drivers/lora/shell.c

```diff
@@ -0,0 +1,247 @@
+			    (int)modem_config.coding_rate + 4);
+		shell_print(shell, "  Preamble length: %" PRIu16,
+			    modem_config.preamble_len);
+
+		return 0;
+	}
+
+	for (int i = 1; i < argc; i += 2) {
+		char *param = argv[i];
+		char *eptr;
+		int value;
+
+		if (i + 1 >= argc) {
+			shell_error(shell, "'%s' expects an argument",
+				    argv[i]);
+			return -EINVAL;
+		}
+
+		value = strtol(argv[i + 1], &eptr, 0);
+		if (*eptr != '\0') {
+			shell_error(shell, "'%s' is not an integer",
+				    argv[i + 1]);
+			return -EINVAL;
+		}
+
+		if (!strcmp("freq", param)) {
+			modem_config.frequency = value;
+		} else if (!strcmp("tx-power", param)) {
+			modem_config.tx_power = value;
+		} else if (!strcmp("bw", param)) {
+			switch (value) {
+			case 125:
+				modem_config.bandwidth = BW_125_KHZ;
+				break;
+			case 250:
+				modem_config.bandwidth = BW_250_KHZ;
+				break;
+			case 500:
+				modem_config.bandwidth = BW_500_KHZ;
+				break;
+			default:
+				shell_error(shell, "Invalid bandwidth: %s",
+					    value);
+				return -EINVAL;
+			}
+		} else if (!strcmp("sf", param)) {
+			if (value < SF_6 || value > SF_12) {
+				shell_error(shell,
+					    "Invalid spreading factor: SF%i",
+					    value);
+				return -EINVAL;
+			}
+
+			modem_config.datarate = value;
+		} else if (!strcmp("cr", param)) {
+			if (value < 5 || value > 8) {
+				shell_error(shell, "Coding rate: 4/%i",
+					    value);
+				return -EINVAL;
+			}
+
+			modem_config.coding_rate = CR_4_5 + value - 5;
+		} else if (!strcmp("pre-len", param)) {
+			modem_config.preamble_len = value;
+		} else {
+			shell_error(shell, "Unknown parameter '%s'", param);
+			return -EINVAL;
+		}
+	}
+
+	return 0;
+}
+
+static int cmd_lora_send(const struct shell *shell,
+			size_t argc, char **argv)
+{
+	int ret;
+	struct device *dev;
+
+	modem_config.tx = true;
+	dev = get_modem(shell, &modem_config);
+	if (!dev) {
+		return -ENODEV;
+	}
+
+	ret = lora_send(dev, argv[1], strlen(argv[1]));
+	if (ret < 0) {
+		shell_error(shell, "LoRa send failed: %i", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int cmd_lora_recv(const struct shell *shell,
+			size_t argc, char **argv)
+{
+	static char buf[0xff];
+	struct device *dev;
+	int ret;
+	s16_t rssi;
+	s8_t snr;
+
+	modem_config.tx = false;
+	dev = get_modem(shell, &modem_config);
+	if (!dev) {
+		return -ENODEV;
+	}
+
+	ret = lora_recv(dev, buf, sizeof(buf), K_FOREVER, &rssi, &snr);
+	if (ret < 0) {
+		shell_error(shell, "LoRa recv failed: %i", ret);
+		return ret;
+	}
+
+	shell_print(shell, "Data: %s", buf);
+	shell_print(shell, "RSSI: %" PRIi16 " dBm, SNR:%" PRIi8 " dBm",
+		    rssi, snr);
+
+	return 0;
+}
+
+static int cmd_lora_test_cw(const struct shell *shell,
+			    size_t argc, char **argv)
+{
+	struct device *dev;
+	int ret;
+	u32_t freq;
+	s8_t power;
+	u16_t duration;
+
+	dev = get_modem(shell, NULL);
+	if (!dev) {
+		return -ENODEV;
+	}
+
+	freq = atoi(argv[1]);
+	power = atoi(argv[2]);
+	duration = atoi(argv[3]);
```

Comment:
In `cmd_lora_conf()` you use `strtol`. I think we should be consistent and use the same here.